test/runtime/ct.go

```diff
@@ -128,254 +145,411 @@ var _ = Describe("RuntimeConntrackTable", func() {
+			{meta[client], meta[server], "80", helpers.IPv6, http},
+			{meta[client], meta[server], "80", helpers.IPv4, http},
+			{meta[client2], meta[server], "80", helpers.IPv6, http},
+			{meta[client2], meta[server], "80", helpers.IPv4, http},
+			{meta[client], meta[server2], "80", helpers.IPv6, http},
+			{meta[client], meta[server2], "80", helpers.IPv4, http},
+			{meta[client2], meta[server2], "80", helpers.IPv6, http},
+			{meta[client2], meta[server2], "80", helpers.IPv4, http},
+		}
+
+		By("Testing if client and client-2 can talk with server and server-2")
+		for _, testCase := range testCombinations {
+			testReach(testCase.src[helpers.Name], testCase.destination[testCase.kind], testCase.dstPort, testCase.mode, BeTrue)
+		}
+
+		epIdentities, err := vm.GetEndpointsIdentityIds()
+		Expect(err).To(BeNil(), "Getting endpoint identities")
+
+		By("Counting number of CT Entries")
+
+		for _, testCase := range testCombinations {
+			dstIP := testCase.destination[testCase.kind]
+			dstPort := testCase.dstPort
+			srcIP := testCase.src[testCase.kind]
+			srcSecID := epIdentities[testCase.src[helpers.Name]]
+
+			data, err := countCTINEntriesOf(dstIP, dstPort, srcIP, srcSecID)
+			Expect(err).To(BeNil(), "Trying to count CT entries of %s and %s", dstIP, srcIP)
+
+			des := fmt.Sprintf("Checking CT entries between %s and %s", dstIP, srcIP)
+			Expect(data).To(BeNumerically(comparator, 1), des)
+		}
+
+		res := vm.PolicyDel("l3-only-policy-server-2")
+		res.ExpectSuccess("Deleting policy `l3-only-policy-server-2`")
+
+		areEndpointsReady := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not in ready state after deleting policy `l3-only-policy-server-2`")
+
+		vm.ExecCilium("endpoint list")
+
+		By("Checking if all server-2 CT IN entries are gone after deleting policy `l3-only-policy-server-2`")
+		for _, testCase := range testCombinations {
+			dstIP := testCase.destination[testCase.kind]
+			dstPort := testCase.dstPort
+			srcIP := testCase.src[testCase.kind]
+			srcSecID := epIdentities[testCase.src[helpers.Name]]
+
+			data, err := countCTINEntriesOf(dstIP, dstPort, srcIP, srcSecID)
+			Expect(err).To(BeNil(), "Trying to count CT entries of %s and %s", dstIP, srcIP)
+
+			wantCTEntries := 0
+			if dstIP != meta[server2][testCase.kind] {
+				wantCTEntries = 1
+			}
+
+			des := fmt.Sprintf("Checking CT entries between %s and %s", dstIP, srcIP)
+			Expect(data).To(BeNumerically(comparator, wantCTEntries), des)
+		}
 
-	AfterEach(func() {
-		vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
-		vm.PolicyDelAll()
-		netcatPort = 11111
 	})
 
-	It("tests conntrack tables between client to server", func() {
-		vm.SetPolicyEnforcement(helpers.PolicyEnforcementAlways)
-		containers(helpers.Create)
-		defer containers(helpers.Delete)
-		vm.WaitEndpointsReady()
+	It("testing conntrack entries clean up with L3-L4 policy", func() {
+
+		By("Installing L3-L4 policy")
+
+		policy := `[{
```

Comment:
I would like to have some indention here. As line 509

Reply:
Done